front/pages/api/v1/w/[wId]/assistant/generic_agents.ts

```diff
@@ -0,0 +1,370 @@
+import type { CreateAgentConfigurationWithDefaultsResponseType } from "@dust-tt/client";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import {
+  buildSelectedEmojiType,
+  makeUrlForEmojiAndBackground,
+} from "@app/components/assistant_builder/avatar_picker/utils";
+import { createGenericAgentConfigurationWithDefaultTools } from "@app/lib/api/assistant/configuration/agent";
+import { withPublicAPIAuthentication } from "@app/lib/api/auth_wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { getFeatureFlags } from "@app/lib/auth";
+import { apiError } from "@app/logger/withlogging";
+import type { WithAPIErrorResponse } from "@app/types";
+import { getLargeWhitelistedModel } from "@app/types";
+
+export const CreateAgentConfigurationWithDefaultsRequestSchema = t.type({
+  name: t.string,
+  description: t.string,
+  instructions: t.string,
+  emoji: t.union([t.string, t.undefined]),
+  pictureUrl: t.union([t.string, t.undefined]),
+  subAgentName: t.union([t.string, t.undefined]),
+  subAgentDescription: t.union([t.string, t.undefined]),
+  subAgentInstructions: t.union([t.string, t.undefined]),
+  subAgentEmoji: t.union([t.string, t.undefined]),
+});
+
+function assistantHandleIsValid(handle: string) {
+  return /^[a-zA-Z0-9_-]{1,30}$/.test(handle);
+}
+
+/**
+ * @swagger
+ * /api/v1/w/{wId}/assistant/generic_agents:
```

Comment:
Are we sure we want this in the user doc already? I would remove the swagger for now so it's "hidden"

Reply:
Also, should we call it global_agents instead? It could even be `/api/v1/w/{wId}/assistant/global`?

Reply:
"generic" feels clearer, because these agents aren't necessarily global